base/util.jl

```diff
@@ -509,6 +509,32 @@ function _kwdef!(blk, params_args, call_args)
     blk
 end
 
+"""
+    @invokelatest f(args...; kwargs...)
+
+Provides a convenient way to call [`Base.invokelatest`](@ref).
+`@invokelatest f(args...; kwargs...)` will simply be expanded into
+`Base.invokelatest(f, args...; kwargs...)`.
+"""
+macro invokelatest(ex)
+    @assert is_expr(ex, :call) "call expression f(args...; kwargs...) should be given"
+
+    f = first(ex.args)
+    args = []
+    kwargs = []
+    for x in ex.args[2:end]
+        if is_expr(x, :parameters)
+            append!(kwargs, x.args)
+        elseif is_expr(x, :kw)
+            push!(kwargs, x)
+        else
+            push!(args, x)
+        end
+    end
```

Comment:
I think the loop is not necessary --- wouldn't just splatting all of `ex.args[2:end]` into the expression work?

Reply:
na, I think the current parser requires the loop to correctly handle keyword arguments, e.g.:
```julia
julia> macro invokelatest(ex)
           @assert is_expr(ex, :call) "call expression f(args...; kwargs...) should be given"
       
           f = first(ex.args)
           args = ex.args[2:end]
           return esc(:($(GlobalRef(Base, :invokelatest))($(f), $(args...))))
       end
@invokelatest (macro with 1 method)

julia> @macroexpand @invokelatest printstyled(stdout, "blue"; color = :blue)
:(Base.invokelatest(printstyled, $(Expr(:parameters, :($(Expr(:kw, :color, :(:blue)))))), stdout, "blue"))

julia> @invokelatest printstyled(stdout, "blue"; color = :blue)
ERROR: syntax: invalid syntax ; color = :blue
Stacktrace:
 [1] top-level scope
   @ none:1
```